From 14d3bba083de2a694c1998e3d3dacd60777e4124 Mon Sep 17 00:00:00 2001
From: Luca Stefani <luca.stefani.ge1@gmail.com>
Date: Tue, 5 Feb 2019 09:41:43 +0100
Subject: [PATCH] Squash of
 github/lineage-16.0(a8e85dbfc)..lineage-16.0-android-9.0.0_r31(f98ecbe6e) (DO
 NOT SUBMIT)

Luca Stefani (1):
 - Merge tag 'android-9.0.0_r31' into lineage-16.0-android-9.0.0_r31

Raman Tenneti (1):
 - AOSP/Email - Second part of the Security Vulnerability fix -   Email
   App: Malicious app is able to compose message with hidden
   attachments and bypass attachments path checks attaching private
   files   from /data/data/com.android.email/*

android-build-team Robot (1):
 - Merge cherrypicks of [5704859, 5705300, 5704034, 5704195, 5705082,
   5704058, 5704059, 5704932, 5705340, 5705341, 5705342, 5705343,
   5705344, 5705361, 5705362, 5705363, 5705364, 5704870, 5704196,
   5705083, 5701785, 5701786, 5701787, 5704035, 5705261, 5705281,
   5704036, 5704037, 5704038, 5704871, 5704933, 5704872, 5705347,
   5705262, 5704934] into pi-qpr2-release

Change-Id: Ia91af80e02dfd0418df0754b3c024b54e9fd1e56

Change-Id: I976189ba92046d0b3d3c3d03a3dc0658aba46e16
---
 AndroidManifest.xml                           | 15 ++++++++-
 .../ComposeActivityEmailExternal.java         | 32 +++++++++++++++++++
 2 files changed, 46 insertions(+), 1 deletion(-)
 create mode 100644 src/com/android/email/activity/ComposeActivityEmailExternal.java

diff --git a/AndroidManifest.xml b/AndroidManifest.xml
index 370250d27..65a1e1a30 100644
--- a/AndroidManifest.xml
+++ b/AndroidManifest.xml
@@ -92,7 +92,7 @@
 
         <uses-library android:name="org.apache.http.legacy" android:required="false" />
         <activity
-            android:name="com.android.email.activity.ComposeActivityEmail"
+            android:name="com.android.email.activity.ComposeActivityEmailExternal"
             android:label="@string/app_name"
             android:documentLaunchMode="intoExisting"
             android:autoRemoveFromRecents="true"
@@ -147,6 +147,19 @@
                 <data android:scheme="mailto" />
             </intent-filter>
 
+        </activity>
+
+        <!--
+            There are 2 ComposeActivityEmail activities (here and above) because one is listening
+             for intents broadcasted internally and the other for those broadcasted from external
+             applications. Refer to b/32068883.
+        -->
+        <activity android:name="com.android.email.activity.ComposeActivityEmail"
+            android:exported="false"
+            android:label="@string/app_name"
+            android:documentLaunchMode="intoExisting"
+            android:autoRemoveFromRecents="true"
+            android:theme="@style/ComposeTheme">
             <intent-filter>
                 <action android:name="com.android.mail.intent.action.LAUNCH_COMPOSE" />
                 <category android:name="android.intent.category.DEFAULT" />
diff --git a/src/com/android/email/activity/ComposeActivityEmailExternal.java b/src/com/android/email/activity/ComposeActivityEmailExternal.java
new file mode 100644
index 000000000..455193bea
--- /dev/null
+++ b/src/com/android/email/activity/ComposeActivityEmailExternal.java
@@ -0,0 +1,32 @@
+/**
+ * Copyright (C) 2018 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.email.activity;
+
+/**
+ * A subclass of {@link ComposeActivityEmail} which is exported for other Android packages to open.
+ */
+public class ComposeActivityEmailExternal extends ComposeActivityEmail {
+
+  /**
+   * Only relevant when WebView Compose is enabled. Change this when WebView
+   * Compose is enabled for Email.
+   */
+  @Override
+  public boolean isExternal() {
+      return false;
+  }
+}
-- 
2.17.1

